src/common/mappers/switchForm.ts

```diff
@@ -1,14 +1,41 @@
 type Args = Pick<SwitchRoleForm, 'csrf' | 'redirect_uri' | '_fromAWSRoleSwitchExtension'>
-
+type AccountRole = {roleName: string, account: string}
 export const mapToSwitchForm = (
   configItem: AWSConfigItem, 
   args: Args,
-): SwitchRoleForm => ({
-  account: configItem.aws_account_id,
-  roleName: configItem.role_name,
-  color: configItem.color?.replace('#', ''),
-  displayName: `${configItem.title} | ${configItem.aws_account_id}`.slice(0, 64),
-  action: 'switchFromBasis',
-  mfaNeeded: '0',
-  ...args,
-});
+): SwitchRoleForm => {
+    
+  const accountAndRoleName = accountAndRoleNameFromConfigItem(configItem);
+  return ({
+    ...accountAndRoleName,
+    color: configItem.color?.replace('#', ''),
+    displayName: `${configItem.title} | ${accountAndRoleName.account}`.slice(0, 64),
+    action: 'switchFromBasis',
+    mfaNeeded: '0',
+    ...args,
+  });
+};
+
+const emptyRoleAccount = {roleName: '', account:''};
+
+export const accountAndRoleNameFromConfigItem = (ci: AWSConfigItem) : AccountRole => {
+
+  if (ci.aws_account_id && ci.role_name) {
+    return {account: ci.aws_account_id, roleName: ci.role_name};
+  }
+  if (ci.role_arn) {
+    return extractAccountAndRoleFromRoleARN(ci.role_arn);
+  }
+  return emptyRoleAccount;
+};
+
+export const extractAccountAndRoleFromRoleARN = (roleArn: string) : AccountRole =>  {
+  const arnRoleRe = /^arn:aws:iam::(?<account>\d{12}):role\/(?<roleName>\w+)/;
+  const match = roleArn.trim().match(arnRoleRe);
+
+  if (match?.groups) {
+    return {roleName : match.groups.roleName, account: match.groups.account};
+  }
+  
+  return emptyRoleAccount;
+};
```

Comment:
we could put this into the mapper to output `role_name` and `aws_account_id`

Reply:
Above this [line](https://github.com/janstuemmel/aws-role-switch/blob/master/src/common/config/mapper.ts#L60) would make sense i think, because when it's doesnt match a arn string, the entry is filtered out...